src/boot/bootctl.c

```diff
@@ -2331,29 +2331,28 @@ static int verb_is_installed(int argc, char *argv[], void *userdata) {
 
 static int parse_timeout(const char *arg1, char16_t **ret_timeout, size_t *ret_timeout_size) {
         char utf8[DECIMAL_STR_MAX(usec_t)];
-        char16_t *encoded;
-        usec_t timeout;
         int r;
 
         assert(arg1);
         assert(ret_timeout);
         assert(ret_timeout_size);
 
-        if (streq(arg1, "menu-force"))
-                timeout = USEC_INFINITY;
-        else if (streq(arg1, "menu-hidden"))
-                timeout = 0;
+        if (STR_IN_SET(arg1, "menu-force", "menu-hidden"))
+                strcpy(utf8, arg1);
```

Comment:
hmpf, strcpy(). that raises all alarm bells...

at the very least the utf8[] array should be declared as:

```c
char utf8[CONST_MAX3(DECIMAL_STR_MAX(usec_t), STRLEN("menu-force"), STRLEN("menu-hidden"));
```

(use CONST_MAX3() and STRLEN() instead of MAX3() and strlen() so this doesn't degrade into a VLA)